components/layout_2020/flow/inline.rs

```diff
@@ -586,20 +619,19 @@ fn layout_atomic(
     ifc: &mut InlineFormattingContextState,
     atomic: &mut IndependentFormattingContext,
 ) {
-    let style = atomic.style();
+    let style = &atomic.style().clone();
     let pbm = style.padding_border_margin(&ifc.containing_block);
     let margin = pbm.margin.auto_is(Length::zero);
     let pbm_sums = &(&pbm.padding + &pbm.border) + &margin;
-    ifc.inline_position += pbm_sums.inline_start;
-    let mut start_corner = Vec2 {
-        block: pbm_sums.block_start,
-        inline: ifc.inline_position - ifc.current_nesting_level.inline_start,
-    };
-    if style.clone_position().is_relative() {
-        start_corner += &relative_adjustement(&style, ifc.containing_block)
-    }
 
-    let fragment = match atomic {
+    let collects_for_nearest_positioned_ancestor = ifc
+        .positioning_context
+        .collects_for_nearest_positioned_ancestor();
+    let mut child_positioning_context =
```

Comment:
Let's make this an `Option<PositioningContext>` that is only set for non-replaced elements. That will avoid the allocation of 2 vectors for all replaced elements.

Reply:
Will do.